cypress/e2e/tests/pages/explorer/namespace-picker.spec.ts

```diff
@@ -0,0 +1,204 @@
+import ClusterDashboardPagePo from '@/cypress/e2e/po/pages/explorer/cluster-dashboard.po';
+import { NamespaceFilterPo } from '@/cypress/e2e/po/components/namespace-filter.po';
+import { WorkloadsPodsListPagePo } from '@/cypress/e2e/po/pages/explorer/workloads-pods.po';
+
+const clusterDashboard = new ClusterDashboardPagePo('local');
+const namespacePicker = new NamespaceFilterPo();
+
+describe('Namespace picker', () => {
+  beforeEach(() => {
```

Comment:
the `goTo` in each spec will reload the page still. the workload page will need a navTo which will 
- use the burger menu to click on the local cluster
- use the side menu to click on workloads

Reply:
I could do this to the first test but I'm hesitant to remove `goTo` from all of the tests as they use the `cy.getRancherResource` custom command.
Omitting the `goTo` from each spec will make the `cy.getRancherResource` custom command obsolete as it relies on the `goTo` to work. That line of code is what makes these test scenarios deterministic and reliable (in this case it resets the namespace picker to its default state); without it, we are running the risk of introducing flaky tests. 
Also, in removing the `goTo` we are making these tests rely on one another which go against Cypress best practices https://docs.cypress.io/guides/references/best-practices#Having-Tests-Rely-On-The-State-Of-Previous-Tests

Reply:
I have tried to use the UI to reset the namespace picker to its default state with the below lines of code however it only works sometimes. cy.getRancherResource() is reliable as it works every time which is why I'm relying on that heavily here.

```
// Reset: clear selection from dropdown menu
    namespacePicker.clearSelectionButton();
    namespacePicker.isChecked('Only User Namespaces');
    namespacePicker.checkIcon().should('have.length', 1);
```

Reply:
I've update the first test to use `navTo` but the remaining tests need the page reload because beforeEach of the tests I'm sending a request to ensure the namespace picker is back in its default state. Without this the test scenarios will be less reliable and cause failures in other specs (i.e. workloads, jobs, and deployments specs which rely on ns picker to be in default state). Also, in removing the `goTo` we are making these tests rely on one another which goes against Cypress best practices https://docs.cypress.io/guides/references/best-practices#Having-Tests-Rely-On-The-State-Of-Previous-Tests.

Reply:
I have tried to use the UI to reset the namespace picker to its default state with the below lines of code however it only works sometimes.
 
```
// Reset: clear selection from dropdown menu
    namespacePicker.clearSelectionButton();
    namespacePicker.isChecked('Only User Namespaces');
    namespacePicker.checkIcon().should('have.length', 1);
```

Reply:
These things are always a balance between tests taking forever to execute (goTos -> page reload) and tests in the same session (navTos --> click to). The latter does have some value in e2e tests, given users won't refresh between actions. 

The tests currently don't rely on each other (the initial state of the filter isn't checked and the first interaction is to open the filter and set a new value). The reset of the filter in `after` also should mean tests outside of this suite aren't affected.

For `namespacePicker.isChecked` it has a `contains`. Is that one that waits for the content or will fail immediately? Given above though this might not be needed?

Reply:
You are correct in that the initial state of the filter isn't checked but it will still fail further down the test. For example, lets say "default" was selected in a previous test (and it didn't get cleared), if the next test expects "default" to be cleared and selects that option, it will be deselected and throw off the test.
This is why it's important to do a reset in `beforeEach` and since i'm using the api to reset the state of the namespace picker, a reload (or goTo) is required for the UI to reflect the default state, otherwise the namespace picker will contain the previous filter state and cause failures within this spec. I can show you what I mean over a call if you would like.

`contains` is the one that waits for the content and fails after some time if content isn't found. I have that there so that we're testing the check mark is being applied to the correct option after it is selected.

Reply:
Ok, i see what you mean. User preferences aren't automatically updated given changes server side.

Can the beforeEach be changed to a navTo & change filter to all user, instead of the api request? That would mean the gotTo in the other tests can be dropped

Reply:
Yes, I updated the beforeEach to use navTo and UI instead of goTo and api to reset to default and removed all goTo's from the tests